dp/cloud/python/magma/test_runner/tests/test_dp_with_orc8r.py

```diff
@@ -39,46 +43,165 @@
+            .with_capabilities() \
+            .with_frequency_preferences() \
+            .with_desired_state() \
+            .with_antenna_gain() \
+            .with_single_step_enabled(True) \
+            .with_serial_number(self.serial_number)
+
+        cbsd_id = self.given_single_step_cbsd_provisioned(builder)
+
+        self._check_provision_logs()
 
         self.delete_cbsd(cbsd_id)
 
     def test_cbsd_unregistered_when_requested_by_desired_state(self):
-        builder = CbsdAPIDataBuilder().with_serial_number(self.serial_number)
-        cbsd_id = self.given_cbsd_provisioned(builder)
+        builder = CbsdAPIDataBuilder() \
+            .with_capabilities() \
+            .with_frequency_preferences() \
+            .with_desired_state() \
+            .with_antenna_gain() \
+            .with_serial_number(self.serial_number)
+        cbsd_id = self.given_multi_step_cbsd_provisioned(builder)
 
         with self.while_cbsd_is_active():
             filters = get_filters_for_request_type('deregistration', self.serial_number)
 
-            builder = builder.with_desired_state('unregistered')
-            self.when_cbsd_is_updated(cbsd_id, builder.build_post_data())
+            builder = builder.with_desired_state(UNREGISTERED)
+            self.when_cbsd_is_updated_by_user(cbsd_id, builder.payload)
 
             # TODO maybe asking for state (cbsd api instead of log api) would be better
             self.then_message_is_eventually_sent(filters)
 
+    def test_cbsd_unregistered_when_enodebd_changes_coordinates(self):
+        builder = CbsdAPIDataBuilder() \
+            .with_capabilities() \
+            .with_frequency_preferences() \
+            .with_desired_state() \
+            .with_antenna_gain(15) \
+            .with_serial_number(self.serial_number) \
+            .with_latitude_deg(100) \
+            .with_longitude_deg(1000) \
+            .with_cbsd_category("b")
+
+        self.given_multi_step_cbsd_provisioned(builder)
+        with self.while_cbsd_is_active():
+            filters = get_filters_for_request_type('deregistration', self.serial_number)
+
+            update_request = EnodebdUpdateCbsdRequest(
+                serial_number=self.serial_number,
+                installation_param=InstallationParam(
+                    antenna_gain=DoubleValue(value=15),
+                    latitude_deg=DoubleValue(value=100.1),
+                    longitude_deg=DoubleValue(value=1000.1),
+                    indoor_deployment=BoolValue(value=False),
+                ),
+                cbsd_category="b",
+            )
+
+            self.when_cbsd_is_updated_by_enodebd(update_request)
+            cbsd = self.when_cbsd_is_fetched(self.serial_number)
+            self.then_cbsd_is(cbsd, builder.with_latitude_deg(100.1).with_longitude_deg(1000.1).payload)
+            self.then_message_is_eventually_sent(filters)
+
+    def test_cbsd_not_unregistered_when_coordinates_change_less_than_10_m(self):
```

Comment:
I'm actually wondering if we need to have integration tests for all such detailed scenarios (maybe UTs are enough)

Reply:
why not, the filter for coordinate diff may work properly, but for some reason the parent method may be giving it wrong info

Reply:
:+1: